pages/example-6-xss/common.tsx

```diff
@@ -0,0 +1,53 @@
+import LinkIfNotCurrent from '../../components/LinkIfNotCurrent';
+import { BlogPost } from '../../database/blogPosts';
+
+type Props =
+  | {
+      error: string;
+    }
+  | {
+      blogPosts: BlogPost[];
+    };
+
+export function CommonContent(props: Props) {
+  return (
+    <>
+      <h1>XSS - getServerSideProps</h1>
+
+      <ul>
+        <li>
+          <LinkIfNotCurrent href="/example-6-xss/vulnerable">
+            Vulnerable
+          </LinkIfNotCurrent>
+        </li>
+        <li>
+          <LinkIfNotCurrent href="/example-6-xss/solution-1">
+            Solution 1
+          </LinkIfNotCurrent>
+        </li>
+        <li>
+          <LinkIfNotCurrent href="/example-6-xss/solution-2">
+            Solution 2
+          </LinkIfNotCurrent>
+        </li>
+        <li>
+          <LinkIfNotCurrent href="/example-6-xss/solution-3">
+            Solution 3
+          </LinkIfNotCurrent>
+        </li>
+      </ul>
+
+      <hr />
+
+      <div>
+        The following blog posts should only be visible for logged-in users.
+        <br />
+        <br />
+        If a user is not logged in, an error message should appear.
```

Comment:
Let's improve the description of the problem that the example is illustrating

I'll take this over

Reply:
Done in 24971a9